inference-engine/src/mkldnn_plugin/nodes/mkldnn_strided_slice_node.cpp

```diff
@@ -0,0 +1,516 @@
+            params.dstDims[1] /= realDstDim;
+            params.srcDims[1] /= realSrcDim;
+            params.dstDims.insert(params.dstDims.begin() + 1, realDstDim);
+            params.srcDims.insert(params.srcDims.begin() + 1, realSrcDim);
+        }
+
+        if (params.dstDims.size() > 2)
+            params.lastDstDim /= newDstDims[secondDim.first];
+    }
+}
+
+void MKLDNNStridedSliceNode::execute(mkldnn::stream strm) {
+    if (params.dstDims.size() > 1 && params.nDimsForWork == 1)
+        stridedSliceV();
+    else
+        stridedSlice();
+}
+
+static inline size_t parallel_init(size_t start, size_t nDims, const SizeVector& dims, SizeVector& indexes) {
+    for (int j = nDims - 1; j >= 0; j--) {
+        indexes[j] = start % dims[j];
+        start = start / dims[j];
+    }
+    return start;
+}
+
+void MKLDNNStridedSliceNode::stridedSliceV() {
+    const uint8_t* srcData = reinterpret_cast<const uint8_t*>(this->getParentEdgeAt(DATA_ID)->getMemoryPtr()->GetPtr()) +
+                             (begin[0] * params.srcStrides[0] + begin[1] * params.srcStrides[1]) * params.dataSize;
+    uint8_t* dstData = reinterpret_cast<uint8_t*>(this->getChildEdgeAt(0)->getMemoryPtr()->GetPtr());
+
+    const size_t dstIdx = params.dstStrides[0] * params.dataSize;
+    const size_t srcIdx = stride[0] * params.srcStrides[0] * params.dataSize;
+    const size_t dstShift = params.dstStrides[1] * params.dataSize;
+    const size_t srcShift = stride[1] * params.srcStrides[1] * params.dataSize;
+
+    if (params.dstDims.size() > 2) {
+        parallel_for2d(params.dstDims[0], params.dstDims[1], [&](const size_t i, const size_t j) {
+            cpu_memcpy(&dstData[i * dstIdx + j * dstShift], &srcData[i * srcIdx + j * srcShift], params.lastDstDim);
+        });
+    } else {
+        parallel_for(params.dstDims[0], [&](const size_t i) {
+            cpu_memcpy(&dstData[i * dstIdx], &srcData[i * srcIdx], params.lastDstDim);
+        });
+    }
+}
+
+void MKLDNNStridedSliceNode::stridedSlice() {
+    const uint8_t* srcData = reinterpret_cast<const uint8_t*>(this->getParentEdgeAt(DATA_ID)->getMemoryPtr()->GetPtr()) +
+            (stride.back() == 1 && stride.size() > 1 ? begin[params.nDimsForWork] * params.srcStrides[params.nDimsForWork] * params.dataSize : 0);
+    uint8_t* dstData = reinterpret_cast<uint8_t*>(this->getChildEdgeAt(0)->getMemoryPtr()->GetPtr());
+
+    parallel_nt(0, [&](const int ithr, const int nthr) {
+        size_t start = 0, end = 0;
+        SizeVector indexes(params.nDimsForWork, 0);
+        splitter(params.workAmount, nthr, ithr, start, end);
+        parallel_init(start, params.nDimsForWork, params.dstDims, indexes);
+
+        size_t srcIdx;
+        getSrcIdx(srcIdx, indexes);
+
+        for (size_t iwork = start, dstIdx = start * params.lastDstDim, i = 1; iwork < end; ++iwork, dstIdx += params.lastDstDim) {
+            cpu_memcpy(&dstData[dstIdx], &srcData[srcIdx], params.lastDstDim);
```

Comment:
dstIdx and srcIdx not depend on the dynamic input and can be calculated on creation and stored in containers.

Reply:
Added src and dst indices calculating in `CreatePrimitive`, thank you!